scripts/targetwindow.lic

```diff
@@ -68,14 +75,18 @@ module TargetWindow
   def self.watch_quickbar
     old_targets = GameObj.targets
     loop {
-      if GameObj.targets.length > 0
+      targets = GameObj.targets.select { |t| !(t.noun =~ @grasp_arms || t.name =~ /^animated /) }
+      if targets.length > 0
         output = "<dialogData id='quick' clear='t'>"
-        output += "<link id='twtotal' value='Total Targets: #{GameObj.targets.delete_if { |t| t.noun =~ @grasp_arms }.length}' cmd='target next' echo='target next'/>"
-        GameObj.targets.each { |t|
-          next if t.noun =~ @grasp_arms
+        output += "<link id='twtotal' value='Total Targets: #{targets.length}' cmd='target next' echo='target next'/>"
+        targets.each { |t|
           target_status = TargetWindow.status_fix(t.status) unless t.status.nil?
           output += "<link id='#{t.id}' value='#{target_status} #{t.name.split.map(&:capitalize).join(' ')}  ' cmd='target ##{t.id}' echo='target ##{t.id}'/>"
         }
+        if CharSettings['show_arms']
```

Comment:
Is checking directly against CharSettings inside a loop bad practice? This could be assigned to a var before the loop starts. Not sure if this is causing extra DB lookups each loop when this value is not likely to change (the only way it could be changed while script is running is via ;force.